### PR TITLE
Extract content on configurations into separate chapter

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

See https://github.com/gradle/dotorg-docs/issues/187 for more information.

### CODE REVIEW

File: subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl [hunk 1]
```diff
@@ -174,6 +174,7 @@
                                 <ul id="managing-dependencies">
                                     <li><a href="/userguide/introduction_dependency_management.html">Introduction to Dependency Management</a></li>
                                     <li><a href="/userguide/declaring_dependencies.html">Declaring Dependencies</a></li>
+                                    <li><a href="/userguide/defining_using_configurations.html">Defining and Using Configurations</a></li>
```
Comment (eriwen) [inline subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl]: Optional: perhaps "Managing Dependency Configurations" would clarify the type of configurations we're talking about.

File: subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+</figure>
++++++
+
+For more examples on the usage of configurations to navigate, inspect and post-process metadata and artifacts of assigned dependencies, see <<working_with_dependencies>>.
+
+=== Defining custom configurations
+
+You can also define configurations yourself, so-called _custom configurations_. A custom configuration is useful for separating the scope of dependencies needed for a dedicated purpose.
+
+Let's say you wanted to declare a dependency on the link:https://tomcat.apache.org/tomcat-9.0-doc/jasper-howto.html[Jasper Ant task] for the purpose of pre-compiling JSP files that should _not_ end up in the classpath for compiling your source code. It's fairly simply to achieve that goal by introducing a custom configuration and using it in a task.
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc]: "fairly simply" => "fairly simple"

File: subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl (hunk 1 above)
Comment (eriwen) [reply subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl]: Also this link does not work, it should be defining_and_using_configurations.html

File: subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
++++++
+<figure xmlns:xi="http://www.w3.org/2001/XInclude">
+    <title>Configurations use declared dependencies for specific purposes</title>
+    <imageobject>
+        <imagedata fileref="img/dependency-management-configurations.png" width="200mm" />
+    </imageobject>
+</figure>
++++++
+
+For more examples on the usage of configurations to navigate, inspect and post-process metadata and artifacts of assigned dependencies, see <<working_with_dependencies>>.
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc]: Can we re-organize/remove or rename working_with_dependencies yet?

File: subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+=== Defining custom configurations
+
+You can also define configurations yourself, so-called _custom configurations_. A custom configuration is useful for separating the scope of dependencies needed for a dedicated purpose.
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc]: "also" is superfluous IMO

File: subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc (hunk 3 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc]: No, this has become a fully-fledged chapter with completely new content. I find the title fitting to the content except if you have better proposal.

File: subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl (hunk 1 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl]: > Optional: perhaps "Managing Dependency Configurations" would clarify the type of configurations we're talking about.

Maybe that makes it more clear that we are talking about configurations in the context of dependency management. done

> Also this link does not work, it should be defining_and_using_configurations.html

done

File: subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc (hunk 2 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc]: done

File: subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc (hunk 4 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/userguide/definingUsingConfigurations.adoc]: done

**Comment by bmuschko** [comment]
@eriwen Addressed your comments in https://github.com/gradle/gradle/pull/4305. Please review that PR.

**Pull request merged**